docs/installation/linux_sources.rst

```diff
@@ -60,3 +60,62 @@ To install eProsima Shapes Demo using colcon, please follow the steps below:
     .. code-block:: bash
 
         ShapesDemo
+
+.. _linux_ros2_compilation:
+
+ROS 2 features compilation
+--------------------------
+
+By default eProsima Shapes Demo can be built and used on a ROS 2 installation as long as an installation of Fast DDS version 2.5.1 or higher is available and a QT5 installation is available.
+The build process will try to locate the `Shapes Demo TypeSupport <https://github.com/eProsima/ShapesDemo-TypeSupport>`_ and, if present, will automatically enable ROS2 features.
+
+eProsima provides a different repos file, `shapes-demo-ros2.repos <https://github.com/eProsima/ShapesDemo/blob/master/shapes-demo-ros2.repos>`_, that will download Shapes Demo and its dependencies, including the Shapes Demo TypeSupport, to your workspace.
+
+To build eProsima Shapes Demo with ROS 2 features enabled, inside a ROS 2 Humble installation follow these steps:
+
+1.  Install the required eProsima Fast DDS dependencies and verify that the system meets the installation requirements.
+    The complete list of requirements and dependencies can be found in the
+    `Fast DDS Linux Installation Manual <https://fast-dds.docs.eprosima.com/en/latest/installation/sources/sources_linux.html>`_.
+    Specifically, follow the steps outlined in sections
+    `Requirements <https://fast-dds.docs.eprosima.com/en/latest/installation/sources/sources_linux.html#requirements>`_
+    and
+    `Dependencies <https://fast-dds.docs.eprosima.com/en/latest/installation/sources/sources_linux.html#dependencies>`_.
+
+2.  Install the ROS 2 development tools (`Colcon <https://colcon.readthedocs.io/en/released/>`_ and
+    `Vcstool <https://pypi.org/project/vcstool/>`_) by executing the following command:
+
+    .. code-block:: bash
+
+           pip install -U colcon-common-extensions vcstool
+
+    .. note::
+
+        If this fails due to an Environment Error, add the ``--user`` flag to the ``pip`` installation
+        command.
+
+3.  Create a `ShapesDemo` directory and download the ROS 2 version of the repos file that will be used to install
+    eProsima Shapes Demo and its dependencies:
+
+    .. code-block:: bash
+
+        mkdir -p ShapesDemo/src && cd ShapesDemo
+        wget https://raw.githubusercontent.com/eProsima/ShapesDemo/master/shapes-demo-ros2.repos
+        vcs import src < shapes-demo-ros2.repos
+
+3.  Build the packages:
+
+    .. code-block:: bash
+
+        colcon build
```

Comment:
This fails to find Asio within a `ros:humble` docker container. Also, I feel we should not have to compile things that ROS 2 already has
```suggestion
        cd ~/shapes_demo_ws
        colcon build
```